homeassistant/components/sunweg/__init__.py

```diff
@@ -113,13 +113,13 @@ def get_data(
             entity_description.name,
         )
         variable = entity_description.api_variable_key
-        previous_metric = entity_description.native_unit_of_measurement
+        previous_unit = entity_description.native_unit_of_measurement
         api_value = self.get_api_value(variable, device_type, inverter_id, deep_name)
         previous_value = self.previous_values.get(variable)
         return_value = api_value
-        if entity_description.api_variable_metric is not None:
+        if entity_description.api_variable_unit is not None:
             entity_description.native_unit_of_measurement = self.get_api_value(
```

Comment:
Side note: The entity description is not meant to change after creating the description instance. It should be constant.

Reply:
The API might change the value unit from kWh to MWh, how should I handle this?

Reply:
If the unit is dynamic and can't be determined ahead of time, I'd not include it in the entity description. Entity descriptions are meant for attributes that are the same.

Alternatively it's possible to copy a dataclass to a new dataclass while replacing an attribute with `dataclasses.replace`.